### ISSUE TITLE
[Bug]: unstable_TreeView The L1 retains select state when user collapses and expands L1

### ISSUE THREAD

**Issue by francinelucca** [carbon-design-system/carbon#11840]
### Package

@carbon/react

### Browser

Chrome, Safari, Firefox

### Package version

1.6.0

### React version

17.0.2

### Description

L1 Top levels should not have a select state until an L2 is selected and you happen to collapse the L1. Then it should have the section to indicate an item “inside” that L1 is active.

** Moving this over from https://github.com/carbon-design-system/carbon-platform/issues/1030

https://user-images.githubusercontent.com/40550942/180251543-875a11c6-c97a-4b40-b2b4-f1338c8d2e47.mov

### Reproduction/example

https://stackblitz.com/edit/github-2mpgdk?file=src/App.jsx

### Steps to reproduce

1 - Select an L2 using keyboard navigation
2 - Collapse L2 parent using keyboard navigation
3 - Expand L2 parent using keyboard navigation, notice L2 is still visually selected
4 - Collapse L2 parent using mouse click
5 - Expand L2 parent using mouse click, notice L2 is no longer visually selected

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by emyarod** [comment]
> L1 Top levels should not have a select state until an L2 is selected and you happen to collapse the L1. Then it should have the section to indicate an item “inside” that L1 is active.

I'm not sure if I fully understand the issue but I think this would be application dependent. There are definitely contexts where the selection state of a parent node would need to be separate from selection state of its child nodes (e.g. a file directory like in VS Code or Finder), and changing the parent node behavior to always reflect the selection state of its children would eliminate that

can you provide some more info on what you expect the behavior would be?

**Comment by francinelucca** [comment]
> I'm not sure if I fully understand the issue but I think this would be application dependent. There are definitely contexts where the selection state of a parent node would need to be separate from selection state of its child nodes (e.g. a file directory like in VS Code or Finder), and changing the parent node behavior to always reflect the selection state of its children would eliminate that

If that is the guidance then we can definitely make custom updates on our app, however:

> can you provide some more info on what you expect the behavior would be?

I think the case here is we'd expect the keyboard behavior and the mouse behavior to behave similarly; currently the L2 stays active when using a keyboard and doesn't stay active when using a mouse, is this the intended behavior?

**Comment by emyarod** [comment]
> I think the case here is we'd expect the keyboard behavior and the mouse behavior to behave similarly; currently the L2 stays active when using a keyboard and doesn't stay active when using a mouse, is this the intended behavior?

just to clarify, this is only when referring to interacting with the toggle right? like a user can independently toggle the caret with the keyboard without affecting selection, but mouse clicks will always affect selection?

I'll have to think about this one some more because there are cases where mouse clicks are always tied to selection+activation, but there are also examples where they aren't

for example in VS Code, mouse actions always overwrite the current selection state and put selection on the parent node

https://user-images.githubusercontent.com/8265238/181596395-876cd720-6891-4db3-8dd9-8be0f24d6cab.mov

macOS Finder will just erase the selection state completely

https://user-images.githubusercontent.com/8265238/181596642-ae4b9266-6dc8-445e-9590-2dd3f0bb8643.mov

even though both of these are multiselect trees, I didn't want to completely remove the selection state for a single select context so I went with the VS Code route

I guess a couple options now are:

* changing the default behavior (so that toggle clicks only affect expansion state)
* adding a new config option that can affect this behavior

I would lean towards the first option if we have to make any change at all, but I need to remember if there were any reasons for not going with this approach in the first place. definitely open to more discussion on this topic

**Comment by tay1orjones** [comment]
> I think the case here is we'd expect the keyboard behavior and the mouse behavior to behave similarly; currently the L2 stays active when using a keyboard and doesn't stay active when using a mouse, is this the intended behavior?

Agree, the interaction should be the same between mouse and keyboard

> even though both of these are multiselect trees, I didn't want to completely remove the selection state for a single select context so I went with the VS Code route

With this it sounds like the actionable item here is we need to update keyboard interaction to move selection state to the parent when the parent is opened/closed. This would align the keyboard interaction to what happens via click/mouse.

**Comment by emyarod** [comment]
It looks like VS Code tree behavior regarding collapsing/expanding nodes with selected child nodes now matches the Carbon tree view behavior, and based on some additional discussion, we will close this issue for now as wontfix

https://github.com/user-attachments/assets/694e0979-ae93-4600-aa35-e5ea4eb8aed8
